api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -703,6 +703,20 @@ private Boolean copyBlob(String bucketName, Blob b) {
       }
     }
 
+    // If a lock is held by another user, log this to establish a rough estimate of how often
+    // locked notebooks are encountered. Note that this only covers locks encountered from the
+    // Workbench - any Jupyter UI-based lock detection does not touch this code path.
+    if (response.getLockExpirationTime() != null && response.getLastLockedBy() != null) {
+      String currentUsername = userProvider.get().getUsername();
+      if (clock.millis() < response.getLockExpirationTime()
+          && !response.getLastLockedBy().equals(currentUsername)) {
+        log.info(
```

Comment:
If it's useful, we can either fire an event metric for this to StackDriver, or add a log-based metric.

Reply:
Nice suggestion, added - PTAL